Background Scripts/Prevent unnecessary notifications from being sent out/Prevent_unnecessary_notifications_from_being_sent_out.js

```diff
@@ -0,0 +1,16 @@
+// Create a GlideRecord object for the 'sys_email' table
+var emailGR = new GlideRecord('sys_email');
+
+// Query for the emails you want to ignore (adjust the query as needed)
+emailGR.addQuery('state', 'ready'); // only those mails which are ready to send
+emailGR.addEncodedQuery("sys_created_onONToday@javascript:gs.beginningOfToday()@javascript:gs.endOfToday()"); // Optional query to set timeline if not required we can comment this.
+emailGR.query();
+
+// Loop through the results and mark them as 'Ignored'
+while (emailGR.next()) {
```

Comment:
Sorry I wasn't entirely clear - when you use updateMultiple(), you don't need to loop through the records - that's the point! You send the query **AND** the update to the database in 1 call! So you're code would like like this:
```javascript
// Create a GlideRecord object for the 'sys_email' table
var emailGR = new GlideRecord('sys_email');

// Query for the emails you want to ignore (adjust the query as needed)
emailGR.addQuery('state', 'ready'); // only those mails which are ready to send
emailGR.addEncodedQuery("sys_created_onONToday@javascript:gs.beginningOfToday()@javascript:gs.endOfToday()"); // Optional query to set timeline if not required we can comment this.

//Set any results to ignored
emailGR.state = "ignored"; //setting state to "ignored"
emailGR.type = 'send-ignored'; // Set the type to 'ignored'
emailGR.updateMultiple(); // Save the changes
``` 

Yeah, you don't even need the `query` in this case, due to what I mentioned above!